core/citrus-spring/src/main/java/com/consol/citrus/common/SpringXmlTestLoader.java

```diff
@@ -43,6 +43,16 @@ public class SpringXmlTestLoader extends DefaultTestLoader implements TestSource
 
     private String source;
 
+    @Override
+    protected void initializeTestRunner() {
+        if (citrusContext == null) {
+            ApplicationContext applicationContext = loadApplicationContext();
+            citrusContext = CitrusSpringContext.create(applicationContext);
```

Comment:
Not sure about this one. The `CitrusContext` should have been injected via `@CitrusResource` annotation at this point. If this is not the case for any reason the `CitrusContext` should be retrieved from `Citrus` instance (also being injected) rather than creating it via `loadApplicationContext()`.

If `CitrusContext` is not set at this point `loadApplicationContext()` will not be able to retrieve the parent application context properly and create an empty application context. I do not think that this is what we want.

So if `CitrusContext` is not set at this point for some reason I would try to retrieve it from `Citrus` instance as it is done in `super.initializeTestRunner()`. Otherwise an error should be thrown as it is already done in `super.initializeTestRunner()`.

WDYT?

Reply:
oh, I see. if `citrusContext` is null at this point, then `applicationContext` will also be `null`. I'll revert this change, then try again to implement the next citrus release into our applications. if it doesn't work, I'll submit another PR.